numpy/lib/tests/test_format.py

```diff
@@ -934,6 +936,19 @@ def test_unicode_field_names(tmpdir):
         with assert_warns(UserWarning):
             format.write_array(f, arr, version=None)
 
+def test_header_growth_axis():
+    for is_fortran_array, dtype_space, expected_header_length in [
+        [False, 23, 128], [False, 24, 192], [True, 24, 128], [True, 25, 192]
+    ]:
+        for size in [10**i for i in range(len(str(2**64 - 1)))]:
+            fp = BytesIO()
+            format.write_array_header_1_0(fp, {
+                'shape': (1, size) if is_fortran_array else (size, 1),
```

Comment:
Fun little fact: In that particular case a fortran array is also C-order.  You could thus (in principle) grow it as either C or Fortran order.  I suspect that is irrelevant in practice though.

Reply:
Indeed, this can happen if one created array headers without creating an actual, physical array. I'll give it a quick fix and replace the 1 by a 2. I mean, we do not need to make tests unnecessarily unrealistic, I totally agree with you.